src/components/query-string.tsx

```diff
@@ -1,46 +1,89 @@
 'use client';
 
 import { ActionIcon, CopyButton, InputWrapper, rem, TextInput, Title, Tooltip } from '@mantine/core';
-import { IconArrowRight, IconCheck, IconCopy } from '@tabler/icons-react';
+import { IconArrowRight, IconCheck, IconCopy, IconRefresh, IconSearch } from '@tabler/icons-react';
 import { useRecoilValue } from 'recoil';
 import { activeTypeParamsState } from '@/state/type-params-state';
 import { buildExportRequestString } from '@/util/exportRequestBuilders';
-import { SearchParamsProps } from '@/app/query-builder/page';
+import { useSearchParams } from 'next/navigation';
+import { SupportedExportTypes } from './query-selector/export-type';
+import { useState } from 'react';
+import Link from 'next/link';
 
 /*
  * Component to visualize the Bulk-export request string.
  */
-export default function QueryString({ searchParams }: { searchParams: SearchParamsProps }) {
+export default function QueryString() {
+  const [contentLocation, setContentLocation] = useState<string | null>();
+  const [status, setStatus] = useState<number>();
   const typeParams = useRecoilValue(activeTypeParamsState);
-  const { exportType, id } = searchParams;
-  const exportRequestString = buildExportRequestString({ exportType: exportType, id, typeParams });
+  const searchParams = useSearchParams();
+  const exportType = searchParams.get('exportType') as SupportedExportTypes;
+  const id = searchParams.get('id');
 
+  const exportRequestString = buildExportRequestString({ exportType, id, typeParams });
+
+  const kickoffRequest = () => {
+    fetch(exportRequestString).then(response => {
+      setContentLocation(response.headers.get('content-location'));
+      setStatus(response.status);
+    });
```

Comment:
```suggestion
    fetch(exportRequestString).then(response => {
      setContentLocation(response.headers.get('content-location'));
      setStatus(response.status);
    })
    .catch(err => {
        // do something to set status to show an error.
    });
```

This could use a `.catch` handler here to handle the errors with kicking off the request.